prediction_market_agent/agents/autogen_general_agent/social_agent.py

```diff
@@ -0,0 +1,200 @@
+    }
+
+
+def build_agents(model: str) -> Dict[AutogenAgentType, autogen.ConversableAgent]:
+    llm_config = build_llm_config(model)
+
+    user_proxy = autogen.UserProxyAgent(
+        name="User",
+        human_input_mode="NEVER",
+        is_termination_msg=lambda x: x.get("content", "").find("TERMINATE") >= 0,
+        code_execution_config={
+            "last_n_messages": 1,
+            "work_dir": "tasks",
+            "use_docker": False,
+        },
+    )
+
+    writer = autogen.AssistantAgent(
+        name="Writer",
+        llm_config=llm_config,
+        system_message="""
+        You are a professional influencer, known for your insightful and engaging tweets.
+        You transform complex concepts into compelling narratives.
+        You should imporve the quality of the content based on the feedback from the user.
+        You must always return only the tweet.
+        """,
+    )
+
+    critic = autogen.AssistantAgent(
+        name="Critic",
+        llm_config=llm_config,
+        system_message="""
+            You are a critic, known for your thoroughness and commitment to standards.
+            Your task is to scrutinize content for any harmful elements or regulatory violations, ensuring
+            all materials align with required guidelines.
+            References to betting and gambling are allowed.
+            """,
+    )
+
+    return {
+        AutogenAgentType.CRITIC: critic,
+        AutogenAgentType.WRITER: writer,
+        AutogenAgentType.USER: user_proxy,
+    }
+
+
+def deduplicate_bets(bets: list[OmenBet]) -> list[OmenBet]:
+    seen_titles = set()
+    unique = []
+    for bet in bets:
+        if bet.title not in seen_titles:
+            unique.append(bet)
+            seen_titles.add(bet.title)
+    return unique
+
+
+def build_social_media_text(
+    model: str,
+) -> str | None:
+    """
+    Builds a tweet based on the five markets on Omen that are closing soonest.
+
+    This function utilizes the writer and critic agents to generate the tweet content. It first initializes the
+    necessary agents based on the provided model. Then, it registers a chat between the writer and critic agents.
+
+    The tweet content is generated using a template that includes questions about each market's title and likelihood.
+    """
+
+    agents = build_agents(model)
+    user_proxy, writer, critic = (
+        agents[AutogenAgentType.USER],
+        agents[AutogenAgentType.WRITER],
+        agents[AutogenAgentType.CRITIC],
+    )
+    user_proxy.register_nested_chats(
+        [
+            {
+                "recipient": critic,
+                "message": reflection_message,
+                "summary_method": "last_msg",
+                "max_turns": 1,
+            }
+        ],
+        trigger=writer,
+    )
+
+    sh = OmenSubgraphHandler()
+    one_month_ago = utcnow() - timedelta(days=30)
+    reference_agent = Web3.to_checksum_address(
+        "0xc918c15b87746e6351e5f0646ddcaaca11af8568"
```

Comment:
Shouldn't this be determined from APIKeys (i.e. from the pkey in the secrets)?

Reply:
This is the public_key from the reference agent - I picked the Think-Thoroughly agent and hardcoded it.
I don't think this is configured in the .env, rather it will be the Farcaster agent private key that will be stored under `BET_FROM_PRIVATE_KEY`

Reply:
Is the reason this agent needs a pkey at all because it needs to pay gas fees to post (i.e. when calling `Warpcast(private_key=...).post_cast`)?

If so, do you think it should have a separate wallet to the agent that it's posting on behalf of? Even if so, this address should be configurable (as another env var in `SocialMediaAPIKeys`, so we can easily deploy multiple `FarcasterAgent` that post on behalf of other agents.

Reply:
No gas fees involved - FacasterAgent needs public key of the "tracked" agent simply for retrieving positions that serve as inspiration for tweet

Reply:
Okay cool. So in that case why not just re-use the `BET_FROM_PRIVATE_KEY` env var to mean the pkey of the agent to post on behalf of?